main.js

```diff
@@ -1,11 +1,372 @@
+  }
+
+  function install(cb) {
+    const target = path.basename(process.execPath);
+    exeSquirrelCommand(['--createShortcut', target], cb);
+  }
+
+  function uninstall(cb) {
+    const target = path.basename(process.execPath);
+    exeSquirrelCommand(['--removeShortcut', target], cb);
+  }
+
+  switch (squirrelCommand) {
+    case '--squirrel-install':
+      install(app.quit);
+      break;
+
+    case '--squirrel-updated':
+      // Always quit when done
+      app.quit();
+      return true;
+
+    case '--squirrel-uninstall':
+      // Always quit when done
+      uninstall(app.quit);
+      return true;
+
+    case '--squirrel-obsolete':
+      // This is called on the outgoing version of your app before
+      // we update to the new version - it's the opposite of
+      // --squirrel-updated
+      app.quit();
+      return true;
+    default:
+      break;
+  }
+
+  return true;
+};
+
+if (handleStartupEvent()) {
+  console.log('OpenBazaar started on Windows...');
+}
+
+// Set daemon binary name
+const daemon = (platform === 'darwin' || platform === 'linux') ? 'openbazaard' : 'openbazaard.exe';
+
+const serverPath = `${__dirname}${path.sep}..${path.sep}openbazaar-go${path.sep}`;
+let serverRunning = false;
+let pendingKill;
+// let startAfterClose;
+
+const startLocalServer = function startLocalServer() {
+  if (fs.existsSync(serverPath)) {
+    if (pendingKill) {
+      pendingKill.once('close', startLocalServer());
+      return;
+    }
+
+    if (serverRunning) return;
+
+    console.log('Starting OpenBazaar Server');
+
+    // const random_port = Math.floor((Math.random() * 10000) + 30000);
+
+    const sub = childProcess.spawn(serverPath + daemon, ['start'], {
+      detach: false,
+      cwd: `${__dirname}${path.sep}..${path.sep}openbazaar-go`,
+    });
+
+    serverRunning = true;
+
+    let stdout = '';
+    let stderr = '';
+    let serverOut;
+
+    const stdoutcallback = function (buf) {
+      console.log('[STR] stdout "%s"', String(buf));
+      stdout += buf;
+      serverOut = `${serverOut}${buf}`;
+    };
+    sub.stdout.on('data', stdoutcallback);
+    const stderrcallback = function stderrcallback(err) {
+      if (err) {
+        console.log(err);
+        return err;
+      }
+      return false;
+    };
+    const stderrcb = function (buf) {
+      console.log('[STR] stderr "%s"', String(buf));
+      fs.appendFile(`${__dirname}${path.sep}error.log`, String(buf), stderrcallback);
+      stderr += buf;
+    };
+    sub.stderr.on('data', stderrcb);
+    const closecallback = function (code) {
+      console.log(`exited with ${code}`);
+      console.log('[END] stdout "%s"', stdout);
+      console.log('[END] stderr "%s"', stderr);
+      serverRunning = false;
+    };
+    sub.on('close', closecallback);
+    sub.unref();
+  } else {
+    mainWindow.webContents.executeJavaScript("console.log('Unable " +
+      "to find openbazaard')");
+  }
+};
+startLocalServer();
+
+// let killLocalServer = function () {
```

Comment:
I'm wondering if we should be commenting out unused pieces of code or omitting them altogether (we could always get them from v1 later, right).

But, I guess it's not so much, so I think it's OK for now. I would just want to limit it should it ever get to be a situation where the file is more commented out stuff than not.